ci: make sure the ci detects testground failure

laurentsenta opened the pull request:
Changes the CI action so it detects errors,

Related issue: https://github.com/testground/testground/issues/1329

## Tasks:

- [x] update the CI to make sure it detects errors,

laurentsenta commented:
@mxinden
I was planning on turning this code into a reusable workflow so that the script is invisible to you (we'd just call a script that does the shell gymnastic in the background, later we'll fix the API and update the action), would that work for you?

The rationale for not changing the testground CLI yet is that I'd like to make the existing code work for us before making new testground releases and since this approach has been used for a while, I hope it'll be enough.

laurentsenta commented on .github/workflows/ci.yml:
```diff
@@ -82,33 +82,32 @@ jobs:
 
       - name: Run testground plan (sdk-rust)
+        id: run
+        uses: laurentsenta/testground-github-actions/run-composition@master
```
please ignore these names for now

mxinden commented:
>> To rephrase the above, how about doing https://github.com/testground/testground/issues/1329 instead of the suggested bash code here?
>
> The rationale for not changing the testground CLI yet is that I'd like to make the existing code work for us before making new testground releases and since this approach has been used for a while, I hope it'll be enough.

In the end this is up to you, thus I am not going to push for either here. Though still, here are my 2 Cents for _Bash then Go vs. Go right away_:

- Every line of bash is one line of bash too much. In my opinion it is very hard to maintain bash code and very easy to shoot oneself in the foot.
- While very important, this effort is not time critical. In other words, I don't  think it matters whether (a) we have a working version with bash and later without or (b) a bit delayed a working version without bash right away.
- Given that we plan to go without bash long term, why not deduplicate work and do it in Go right away?
- Priorities change, especially at Protocol Labs. While I hope for the amount of resources spend on Testground to increase, there can always be a sudden change in direction. In case Testground ever gets de-prioritized it would be a bummer to be left with a lot of Bash scripts.

This is not directed at your Bash skills in particular, but Bash in general. (I actually trust your Bash code a lot more than I would ever trust my own.)

mxinden commented:
Thanks for tackling https://github.com/testground/testground/pull/1345. Let me know once this is ready for another review @laurentsenta.

laurentsenta commented:
@mxinden Should be the final form, with the `--wait` I removed the scripts (we don't have long poll yet, I hope that will be enough since the jobs are rather short and running locally).

I'm investigating the example failing with:

```
Jun 21 09:13:22.809601	INFO	run finished with outcome = failure (single:0/1)	{"run_id": "caoolqvs6j5ud4h9elsg", "plan": "sdk-rust", "case": "example", "runner": "local:docker", "instances": 1}
```

mxinden reviewed (approved):
Looks good from what I can tell.

@laurentsenta is this ready to merge?

laurentsenta commented:
@mxinden yes, we're good to go,

btw, I never addressed your comment about bash directly; it triggered a few discussions outside Github. We decided to follow your feedback and implement features "the right way" immediately, even if it takes more time to provide them. Sorry for the lack of direct response and thanks for sharing your notes!

mxinden commented:
Thanks @laurentsenta for the help here. Great to have your support.
